pkgs/development/compilers/dotnet/wrapper.nix

```diff
@@ -22,7 +22,14 @@
 type: unwrapped:
 stdenvNoCC.mkDerivation (finalAttrs: {
   pname = "${unwrapped.pname}-wrapped";
-  inherit (unwrapped) version meta;
+  inherit (unwrapped) version;
+
+  meta = {
+    description = "${unwrapped.meta.description} (wrapper)";
+    maintainers = with lib.maintainers; [ corngood ];
```

Comment:
It should inherit the maintainers from the unwrapped as well shouldn't it?

Reply:
That's sort of what I was getting at with my top level comment.  Why shouldn't it be the maintainers of the wrapper itself?

Reply:
I mean, you can be, but the wrapped package is also affecting the e usage, so their maintainers should be preserved (even though it won't change much as currently you're the maintainer for both).

Reply:
I don't have a good answer for that, which is why I created this draft PR for discussion.  It looks like most other wrappers err on the side of inheriting things.

Reply:
I think to address the original problem we could inherit from meta but remove things like vulnerabilities, deprecations and etc.

Unless if @Atemu has some documentation, code comment, discourse post, matrix chat or something similar saying that wrappers shouldn't inherit meta like that, considering that all other wrappers seem to do it.

Reply:
This isn't any sort of standard or documented best practice, just my opinion.

I hold this opinion precisely because of cases like the linked one where passing through all of meta causes issues. There are further issues such as `meta.position` and who knows what other magic is hidden in meta.

I think other wrappers should also only inherit what is acually necessary.  
Just because it's the predominant pattern, that doesn't mean it's good.

Reply:
> Just because it's the predominant pattern, that doesn't mean it's good.

I tend to agree, which is why this PR goes completely the other way.  IMO the sanest thing to do is just treat the wrapper as its own package and only inherit what's needed.  Otherwise we have to define exactly what a wrapper is, and I don't think it's going to be clear in all cases.

What's your take on `maintainers` specifically?

Reply:
> I think other wrappers should also only inherit what is acually necessary.  
> Just because it's the predominant pattern, that doesn't mean it's good.

Yes, it doesn't, but it gets used as reference for new things constantly and provides a certain measure of standardization.
If this wasn't an issue for other compilers which are used much more widely than .NET I don't see why this should be an issue for us or something addressed at a higher level across all of nixpkgs.

Users are putting themselves at risk by using .NET 6 and/or 7. I don't think this should be easy nor painless to do. It should cause as big of an annoyance as possible so they actually think about what they're doing and maintainers actually do something about it.

Reply:
I replied at top level.  I think we should keep this thread for discussion of `maintainers`.

Reply:
> If this wasn't an issue for other compilers which are used much more widely than .NET I don't see why this should be an issue for us

Because dotnet is the only compiler I know of in Nixpkgs that gets EOL'd via insecure.

Btw, why is the compiler EOL'd anyways? Isn't only the runtime relevant for security? Not to deep into dotnet to know the details here.

> Users are putting themselves **and others** (on their local network, chat apps, social networks, shared devices and more) at risk by using .NET 6 and/or 7. I don't think this should be easy nor painless to do. It should cause as big of an annoyance as possible so they actually think about what they're doing and maintainers actually do something about it.

Having to put it in permittedInsecure is annoyance enough. Anything past that is *extremely* unnecessary.